src/access/wac.test.ts

```diff
@@ -1670,3 +1675,558 @@ describe("getGroupAccessAll", () => {
+    );
+
+    expect(newAccess).toStrictEqual({
+      read: false,
+      append: false,
+      write: false,
+      control: false,
+    });
+  });
+
+  it("throws if the access being set can't be expressed in WAC", async () => {
+    const resource = getMockDataset(
+      "https://some.pod/resource",
+      "https://some.pod/resource.acl"
+    );
+    await expect(
+      setAgentResourceAccess(resource, "https://some.pod/profile#agent", ({
+        controlRead: true,
+        controlWrite: undefined,
+      } as unknown) as WacAccess)
+    ).rejects.toThrow(
+      "For WAC resources, controlRead and controlWrite must be equal."
+    );
+  });
+
+  it("returns null if the ACL cannot be written back", async () => {
+    const aclResource = addMockAclRuleQuads(
+      getMockDataset("https://some.pod/resource.acl"),
+      "https://some.pod/profile#agent",
+      "https://some.pod/resource",
+      { read: false, append: false, write: false, control: false },
+      "resource"
+    );
+
+    const mockFetch = jest
+      .fn(window.fetch)
+      // The resource ACL is available...
+      .mockResolvedValueOnce(
+        mockResponse(await triplesToTurtle(Array.from(aclResource)), {
+          status: 200,
+          url: "https://some.pod/resource.acl",
+        })
+      )
+      // Cannot save the ACL
+      .mockResolvedValueOnce(
+        mockResponse("Not authorized", {
+          status: 403,
+          url: "https://some.pod/.acl",
+        })
+      );
+
+    const resource = getMockDataset(
+      "https://some.pod/resource",
+      "https://some.pod/resource.acl"
+    );
+
+    const result = await setAgentResourceAccess(
+      resource,
+      "https://some.pod/profile#agent",
+      {
+        read: true,
+        append: undefined,
+        write: undefined,
+        controlRead: undefined,
+        controlWrite: undefined,
+      },
+      {
+        fetch: mockFetch,
+      }
+    );
+    expect(result).toBeNull();
+  });
+
+  it("calls the underlying WAC API functions", async () => {
+    const mockFetch = jest.fn(window.fetch).mockResolvedValue(
+      mockResponse("", {
+        status: 200,
+        url: "https://some.pod/resource.acl",
+      })
+    );
+
+    const wacModule = jest.requireActual("../acl/agent") as {
+      setAgentResourceAccess: () => Promise<AgentAccess>;
+      saveAclFor: () => Promise<AclDataset>;
+    };
+    const setAgentResourceAccessWac = jest.spyOn(
+      wacModule,
+      "setAgentResourceAccess"
+    );
+
+    const resource = getMockDataset(
+      "https://some.pod/resource",
+      "https://some.pod/resource.acl"
+    );
+    await setAgentResourceAccess(
+      resource,
+      "https://some.pod/profile#agent",
+      {
+        read: undefined,
+        append: true,
+        write: undefined,
+        controlRead: undefined,
+        controlWrite: undefined,
+      },
+      {
+        fetch: mockFetch,
+      }
+    );
+    expect(setAgentResourceAccessWac).toHaveBeenCalled();
```

Comment:
A check for `saveAclFor` would also be appropriate here, I think?

Reply:
Oops, I initially half-added this test, then added another test to check that if the save failed we returned `null` and decided the former was no longer useful, but did not remove it properly ^^. But yeah, only testing the failure case isn't good either. Fixed in 59e2bf3513e8fe318de5c284bbd870dfae3e86fa.